torchvision/prototype/datasets/_builtin/imagenet.py

```diff
@@ -141,10 +148,13 @@ def _make_datapipe(
                 devkit_dp,
                 images_dp,
                 key_fn=getitem(0),
-                ref_key_fn=self._val_image_key,
+                ref_key_fn=self._val_test_image_key,
                 buffer_size=INFINITE_BUFFER_SIZE,
             )
             dp = Mapper(dp, self._collate_val_data)
+        else:  # config.split == "test"
+            dp = Shuffler(images_dp)
```

Comment:
No `buffer_size=INFINITE_BUFFER_SIZE` needed here?

Reply:
No, that is needed here. Good catch!